core/math/src/main/java/org/arakhne/afc/math/geometry/d2/afp/BasicPathShadow2afp.java

```diff
@@ -184,19 +188,15 @@ private void discretizePathIterator(
             return;
         }
         PathElement2afp element;
-
         element = pi.next();
         if (element.getType() != PathElementType.MOVE_TO) {
             throw new IllegalArgumentException(Locale.getString(Path2afp.class, "E1")); //$NON-NLS-1$
         }
-
-        Path2afp<?, ?, ?, ?, ?, ?> localPath;
         double movx = element.getToX();
         double movy = element.getToY();
         double curx = movx;
         double cury = movy;
-        double endx;
-        double endy;
+        this.coordinateParam = new BasicPathShadow2afp.CoordinatesParam(x1, y1, x2, y2, curx, cury);
```

Comment:
I should have thought  that. Now I have changed it as it should be.

On Fri, Jun 24, 2016 at 9:54 PM, Stéphane Galland <notifications@github.com>
wrote:

> In
> core/math/src/main/java/org/arakhne/afc/math/geometry/d2/afp/BasicPathShadow2afp.java
> <https://github.com/gallandarakhneorg/afc/pull/156#discussion_r68445485>:
>
> >          double movx = element.getToX();
> >          double movy = element.getToY();
> >          double curx = movx;
> >          double cury = movy;
> > -        double endx;
> > -        double endy;
> > +        this.coordinateParam = new BasicPathShadow2afp.CoordinatesParam(x1, y1, x2, y2, curx, cury);
>
> The CoordinateParam should be creating inside the loop. Indeed curx and
> cury are changing over the loops.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gallandarakhneorg/afc/pull/156/files/50728fb568bd50071b22352778131359a9506fc8#r68445485>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AR09BD20D3FYYAFWhP9xbssI4nZjOm9Aks5qPCf4gaJpZM4I9Db2>
> .
>